### ISSUE TITLE
Add Python str and/or repr for spatial vectors

### ISSUE THREAD

**Issue by RussTedrake** [RobotLocomotion/drake#15888]
For many of our core quantities, I get beautiful print overloads, even in python (thanks to @EricCousineau-TRI 's TemplateClass magic):
```
>> print(RigidTransform())
RigidTransform(
  R=RotationMatrix([
    [1.0, 0.0, 0.0],
    [0.0, 1.0, 0.0],
    [0.0, 0.0, 1.0],
  ]),
  p=[0.0, 0.0, 0.0],
)
```
I would like to have the analogous capability when I use spatial vector quantities.  Right now I get, e.g.
```
>> print(SpatialForce())
<pydrake.multibody.math.SpatialForce_[float] object at 0x13e63ddf0>
```
If you simply implement `operator<<` in the c++ classes, then I believe TemplateClass will do the rest.  My dynamics-related notebooks in teaching would be much more beautiful if we implement this simple feature.

**Comment by jwnimmer-tri** [comment]
For `RotationMatrix` and `RigidTransform`, the `__repr__` comes from Python code, not `operator<<`:

https://github.com/RobotLocomotion/drake/blob/2069551812f9c0bb7c98e03ea49e749d9784e239/bindings/pydrake/_math_extra.py#L118-L141

Doing it in Python means that we can emit valid (eval-able) Python code for `__repr__`, instead of what Eigen provides for C++ output operators, which is not generally valid Python code.

That doesn't necessarily mean that `SpatialForce` must do it the same way, but I wanted to correct the misunderstanding in the top post.

**Comment by jwnimmer-tri** [comment]
Fixed in #18112.
